test/plugin_test.py

```diff
@@ -1473,6 +1475,96 @@ def test_invalid_uri(self):
         except lsm.LsmError as le:
             self.assertTrue(le.code == lsm.ErrorNumber.INVALID_ARGUMENT)
 
+    def test_battery_list(self):
+        for s in self.systems:
+            cap = self.c.capabilities(s)
+            if supported(cap, [Cap.BATTERIES]):
+                self.c.batteries()
+
+    def test_volume_cache_info(self):
+        flag_tested = False
+        flag_created = False
+        for s in self.systems:
+            cap = self.c.capabilities(s)
+            if supported(cap, [Cap.VOLUME_CACHE_INFO]):
+                lsm_vols = self.volumes(system_id=s.id)
+                if len(lsm_vols) == 0:
+                    lsm_vol = self._volume_create(s.id)[0]
+                    flag_created = true
+                else:
+                    lsm_vol = lsm_vols[0]
+                cache_info = self.c.volume_cache_info(lsm_vol)
+                self.assertTrue(len(cache_info) == 5)
+                if flag_created:
+                    self._volume_delete(lsm_vol)
+                flag_tested = True
+
+        if flag_tested is False:
+            self._skip_current_test(
+                "Skip test: current system does not support "
+                "volume cache info query")
+
+    def test_volume_cache_set(self):
+        flag_tested = False
+        for s in self.systems:
+            cap = self.c.capabilities(s)
```

Comment:
Check capabilities for all operations you need for test, volume create/delete etc.